ProjectionParameters.h

```diff
@@ -0,0 +1,274 @@
+//####################################################################
+//
+//    FILENAME:          ProjectionParameters.h
+//
+//    CLASSIFICATION:    Unclassified
+//
+//    DESCRIPTION:
+//
+//    Header for object space single image projection parameters.
+//
+//    LIMITATIONS:       None
+//
+//
+//    SOFTWARE HISTORY:
+//     Date          Author    Comment
+//     -----------   ------    -------
+//     31-March-2022    Eugene Rose CCB Change CSM 3.0.4
+//
+//    NOTES:
+//
+//#####################################################################
+
+#ifndef __PROJECTIONPARAMETERS_H
+#define __PROJECTIONPARAMETERS_H
+
+#include "csm.h"
+#include "ObjectSpace.h"
+//>
+// need ObjectSpace.h for ObjectSpaceCoordinate etc.
+//<
+
+namespace csm
+{
+    
+enum class ProjectionGeometryType
+{
+    RANGE,
+    PLANE,
+    RANGECOVAR,
+    PLANECOVAR,
+    UNSPECIFIED
+};
+//>
+// This enum class identifies the projection parameters type.
+//<
+
+// ************************************************************
+// ProjectionParameters virtual base class
+// ************************************************************
+class CSM_EXPORT_API ProjectionParameters
+{
+
+public:
+    virtual ~ProjectionParameters() {};
+    //> 
+    // virtual destructor provides something for base class destructors to call on.
+    //<
+
+    virtual ProjectionGeometryType getType() const = 0;
+    //> 
+    // returns the type enum parameter
+    //<
+protected:
+    ProjectionParameters() {};
+    //> 
+    // the default constructor does nothing
+    //<
+
+};
+
+// ************************************************************
+// ProjectionParametersCovar virtual base class
+// ************************************************************
+class CSM_EXPORT_API ProjectionParametersCovar
+{
+
+public:
+    virtual ~ProjectionParametersCovar() {};
+    //> 
+    // virtual destructor provides something for base class destructors to call on.
+    //<
+
+    virtual ProjectionGeometryType getType() const = 0;
+    //> 
+    // returns the type enum parameter
+    //<
+protected:
+    ProjectionParametersCovar() {};
+    //> 
+    // the default constructor does nothing
+    //<
+
+};
+
+class CSM_EXPORT_API RangeParameters : public ProjectionParameters
+{
+public:
+    RangeParameters();
+    //>
+    // default constructor sets range to zero and also
+    // sets m_type to RANGE
+    //<
+
+    RangeParameters(double range);
+    //>
+    // initializes with the given range in meters
+    //<
+
+    ProjectionGeometryType getType() const;
+    //>
+    // get the geometry type (enum)
+    //<
+
+    const double &getRange() const;
```

Comment:
It's a little inconsistent that the "get" method for the range returns a const reference but the "set" method does not take one.  In this case, since the type is a POD type (`double`), I don't think there is much advantage to doing the const reference.  So either way is fine, but we should be consistent.

A style note: I personally prefer the `&` (or `*` for pointers) to be on the data type, not the name (parameter name or function name).  I realize that it is more /technically/ correct to put them on the name, but it makes it easier for me to think of them modifying the type.  And it only really matters if you are declaring multiple variables in a single statement like `int* a, b` where in this case `a` would be a pointer and `b` would not.  Since that doesn't apply to function names or parameters, you should put the modifier closer to the type name.